File: src/main/java/alfheim/common/entity/boss/ai/AIInvul.java
package alfheim.common.entity.boss.ai;

import alexsocol.asjlib.ASJUtilities;
import alfheim.common.entity.boss.EntityFlugel;

public class AIInvul extends AIBase {

	public AIInvul(EntityFlugel flugel, AITask task) {
		super(flugel, task);
	}
	
	@Override
	public boolean isInterruptible() {
		return false;
	}
	
	@Override
	public void startExecuting() {
		flugel.setAITaskTimer(flugel.SPAWN_TICKS);
	}
	
	@Override
	public boolean continueExecuting() {
		flugel.setHealth(flugel.getHealth() + (flugel.getMaxHealth() - 1F) / flugel.SPAWN_TICKS);
		flugel.motionX = flugel.motionY = flugel.motionZ = 0;
		return canContinue();
	}
	
	@Override
	public void resetTask() {
		flugel.dropState();
	}
}

File: src/main/java/alfheim/common/entity/boss/ai/AIInvul.kt
package alfheim.common.entity.boss.ai;

import alfheim.common.entity.boss.EntityFlugel

class AIInvul(flugel: EntityFlugel, task: AITask) : AIBase(flugel, task) {

	override fun isInterruptible(): Boolean = false

	override fun startExecuting() = flugel.setAITaskTimer(EntityFlugel.SPAWN_TICKS)

	override fun continueExecuting(): Boolean {
		flugel.setHealth(flugel.getHealth() + (flugel.getMaxHealth() - 1f) / EntityFlugel.SPAWN_TICKS)
		flugel.motionX = 0.0
		flugel.motionY = 0.0
		flugel.motionZ = 0.0
		return canContinue()
	}

	override fun resetTask() = flugel.dropState()
}